src/main/java/jenkins/scm/api/SCMFileSystem.java

```diff
@@ -363,6 +363,31 @@ public static boolean supports(@NonNull SCMSource source) {
         return false;
     }
 
+    /**
+     * Given a {@link SCMSourceDescriptor} this method will check if there is at least one {@link SCMFileSystem} provider
+     * capable of being instantiated from the descriptor's {@link SCMSource}. Returning {@code true} does not mean that
+     * {@link #of(SCMSource, SCMHead, SCMRevision)} will be able to instantiate a {@link SCMFileSystem} for any specific
+     * {@link SCMHead} or {@link SCMRevision}, rather returning {@code false} indicates that there is absolutely no point
+     * in calling {@link #of(SCMSource, SCMHead, SCMRevision)} as it will always return {@code null}.
+     *
+     * @param descriptor the {@link SCMSourceDescriptor}.
+     * @return {@code true} if {@link #of(SCMSource, SCMHead)} / {@link #of(SCMSource, SCMHead, SCMRevision)} could
+     * return a {@link SCMFileSystem} implementation, {@code false} if {@link #of(SCMSource, SCMHead)} /
+     * {@link #of(SCMSource, SCMHead, SCMRevision)} will always return {@code null} for the supplied {@link SCMSource}.
+     * @since 2.0
+     */
+    public static boolean supports(@NonNull SCMSourceDescriptor descriptor) {
```

Comment:
If we need this for the `SCMSource`'s `Descriptor` then we need it for the `SCM`'s `Descriptor` too

Reply:
Symmetry.

Plus I can think of quite a few SCM's that would know the answer as being `true` out of the box:
* Subversion
* Accurev (if they ever implement scm-api)

It's really only Git/Mercurial that need to know more about how the repo is hosted before a good answer can be provided... and for that matter same answer applies to `GitSCMSource`

Reply:
Alrightie!